client/jetpack-connect/auth-logged-in-form.jsx

```diff
@@ -172,32 +172,46 @@ class LoggedInForm extends Component {
 
 	redirect() {
 		const { goBackToWpAdmin, redirectAfterAuth } = this.props;
-		const { queryObject } = this.props.authorizationData;
+		const { from } = this.props;
 
-		if (
-			this.props.isSSO ||
-			this.props.isWoo ||
-			this.isFromJpo( this.props ) ||
-			this.shouldRedirectJetpackStart( this.props )
-		) {
+		if ( this.isSso() || this.isWoo() || this.isFromJpo() || this.shouldRedirectJetpackStart() ) {
 			debug(
 				'Going back to WP Admin.',
 				'Connection initiated via: ',
-				queryObject.from,
+				from,
 				'SSO found:',
-				this.props.isSSO
+				this.isSso()
 			);
 			goBackToWpAdmin( redirectAfterAuth );
 		} else {
 			page.redirect( this.getRedirectionTarget() );
 		}
 	}
 
-	isFromJpo( props ) {
-		return startsWith( get( props, [ 'authorizationData', 'queryObject', 'from' ] ), 'jpo' );
+	isFromJpo( { from } = this.props ) {
+		return startsWith( from, 'jpo' );
 	}
 
-	shouldRedirectJetpackStart( { partnerId } ) {
+	/**
+	 * Check whether this a valid authorized SSO request
+	 *
+	 * @param  {?string} props.from            Where is the request from
+	 * @param  {?number} props.queryDataSiteId Remote site ID
+	 * @return {boolean}                       True if it's a valid SSO request otherwise false
+	 */
+	isSso( { from, queryDataSiteId } = this.props ) {
+		const cookies = cookie.parse( document.cookie );
+		const jetpack_sso_approved = cookies.jetpack_sso_approved
+			? parseInt( cookies.jetpack_sso_approved, 10 )
+			: null;
+		return 'sso' === from && !! queryDataSiteId && queryDataSiteId === jetpack_sso_approved;
+	}
+
+	isWoo( { from } = this.props ) {
+		return includes( [ 'woocommerce-setup-wizard', 'woocommerce-services' ], from );
```

Comment:
@marcinbot Thanks for the feedback (https://github.com/Automattic/wp-calypso/pull/19956#issuecomment-347905765).

Are you saying that this string should be changed from `woocommerce-services` to `woocommerce-services-auto-authorize` ?

That string is not introduced in this PR, so that flow isn't working in production 🤔